format/materialized-view-spec.md

```diff
@@ -0,0 +1,132 @@
+<!--
+ - Licensed to the Apache Software Foundation (ASF) under one or more
+ - contributor license agreements.  See the NOTICE file distributed with
+ - this work for additional information regarding copyright ownership.
+ - The ASF licenses this file to You under the Apache License, Version 2.0
+ - (the "License"); you may not use this file except in compliance with
+ - the License.  You may obtain a copy of the License at
+ -
+ -   http://www.apache.org/licenses/LICENSE-2.0
+ -
+ - Unless required by applicable law or agreed to in writing, software
+ - distributed under the License is distributed on an "AS IS" BASIS,
+ - WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ - See the License for the specific language governing permissions and
+ - limitations under the License.
+ -->
+
+# Iceberg Materialized View Spec
+
+## Background and Motivation
+Iceberg views are a powerful tool to abstract complex queries and share them among different engines.
+However, such views are not materialized by default, which means that they are re-computed every time they are queried.
+This can be inefficient for complex queries that are computed frequently.
+Iceberg Materialized views are a way to store the results of an Iceberg view to reuse the computation in subsequent queries.
+
+## Goals 
+The goal of this spec is to define the metadata associated with materialized views in Iceberg.
+Such metadata allows creating and querying Iceberg materialized views across different engines.
+
+## Specification
+A materialized view is an Iceberg view with a respective Iceberg table that stores the results of the view query.
+An Iceberg view is considered a materialized view if it has the `materialized.view` property set to `true`.
+A materialized view must also reference the storage table identifier in its `materialized.view.storage.table` property.
+
+The specification for the materialized view properties on the view is as follows:
+| Property name                              | Description                                                             |
+|--------------------------------------------|-------------------------------------------------------------------------|
+| `materialized.view`                | This property is used to mark whether a view is a materialized view. If set to `true`, the view is treated as a materialized view.|
```

Comment:
Why not just 'materialized'?  (view seems redundant)

Also, in @jankaul's proposal https://docs.google.com/document/d/1UnhldHhe3Grz8JBngwXPA6ZZord1xMedY5ukEhZYF-A/edit#heading=h.a0i0qwejiore  it looks like it was a single field (materialization), and it was a MV if the field is not null?

Reply:
I think the consensus around Iceberg property name is that `.` represents levels, and `-` represents phrases, so I think at least it should be named as `materialized-view.enabled`, `materialized-view.storage-table`, etc. based on my understanding.